### PR TITLE
Ripemd-160 hash function and precompile

### PR DESCRIPTION
**Pull request by edfelten**
* Implement `ripemd160f` instruction (RIPEMD-160 compression function) instruction in Rust AVM and Mini compiler
* Add RIPEMD-160 functionality to the ArbOS library
* Support RIPEMD-160 EVM precompile at address 0x03
* Testing for library and precompile

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,3 +34,4 @@ ethers-signers = "0.1.3"
 keccak="0.1.0"
 bn-plus = "0.4.4"
 rust-crypto="^0.2"
+ripemd160="0.9.1"
```
Comment (hkalodner) [inline Cargo.toml:37]: I'm guessing this is the crate that `ripemd160port.rs` came from, but I don't see any actual usages of this crate in the code. If there aren't any, this should probably be removed as a dependency from the Cargo file

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: src/run/ripemd160port.rs [hunk 2]
```diff
@@ -0,0 +1,295 @@
+use core::convert::TryInto;
+
+pub const DIGEST_BUF_LEN: usize = 5;
+pub const WORK_BUF_LEN: usize = 16;
+
+//type Block = GenericArray<u8, U64>;
+type Block = Vec<u8>;
+
+macro_rules! round(
```
[the remark this answers is not in the document]
Comment (edfelten) [reply src/run/ripemd160port.rs:13]: Added a pointer to the source.

File: Cargo.toml (hunk 1 above)
Comment (edfelten) [reply Cargo.toml]: Removed

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**